browser/resources/settings/email_aliases_page/content/email_aliases_modal.tsx

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import Button from "@brave/leo/react/button"
+import Icon from "@brave/leo/react/icon"
+import { getLocale } from '$web-common/locale'
+import formatMessage from '$web-common/formatMessage'
+import * as React from 'react'
+import { ViewState, ViewMode, MappingService, MAX_ALIASES } from "./types"
+import { Col, Row } from "./style"
+import Input, { InputEventDetail } from "@brave/leo/react/input"
+import styled from "styled-components"
+
+const Modal = styled(Col)`
+  border-radius: var(--cr-card-border-radius);
+  background-color: white;
+  z-index: 2;
+  border: none;
+  opacity: 100%;
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%,-50%);
+  justify-content: flex-start;
+`
+
+const InnerModal = styled.div`
+  width: 42em;
+  margin: 1em 2em;
+`
+
+const ModalSectionCol = styled(Col)`
+  margin: 1em 0em;
+  & h3 {
+    margin: 0.25em;
+  }
+  & leo-input {
+    margin: 0.25em 0em;
+  }
+`
+
+const ButtonRow = styled(Row)<{ bubble: boolean }>`
+  justify-content: ${props => props.bubble ? 'space-between' : 'end'};
+  margin: 1em 0em;
+  & leo-button {
+    flex-grow: 0;
+  }
+  & span {
+    display: flex;
+    flex-direction: row;
+    column-gap: 1em;
+  }
+`
+
+const CloseButton = styled.span`
+  cursor: pointer;
+  position: absolute;
+  top: 1.75em;
+  right: 1.75em;
+`
+
+const GeneratedEmailContainer = styled(Row)`
+  font-size: 135%;
+  background-color:var(--leo-color-neutral-variant-10);
+  border-radius: 0.5em;
+  padding: 0em 0em 0em 0.75em;
+  margin: 0.25em 0em;
+  justify-content: space-between;
+  height: 2.6em;
+  & leo-button {
+    flex-grow: 0;
+  }
+`
+
+const ButtonWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  @keyframes spin {
+    0% { transform: rotate(0deg); }
+    100% { transform: rotate(360deg); }
+  }
+  & .waiting {
+    animation: spin 1s linear infinite;
+  }
+  width: 4em;
+  height: 100%;
+`
+
+const onEnterKey = (onSubmit: () => void) =>
+  (e: InputEventDetail) => {
+    const innerEvent = e.innerEvent as unknown as KeyboardEvent
+    if (innerEvent.key === 'Enter') {
+      onSubmit()
+    }
+  }
+
+const RefreshButton = ({ onClicked }: { onClicked: () => Promise<void> }) => {
+  const [waiting, setWaiting] = React.useState(false)
+  const onClick = async () => {
+    setWaiting(true)
+    await onClicked()
+    setWaiting(false)
```

Comment:
not sure but you might want to try/finally setWaiting 
```suggestion
    try {
        await onClicked()
    } finally {
        setWaiting(false)
    }
```

Reply:
fixed